Sample.BasicComConnector/NewSimRoutine.cs

```diff
@@ -1,82 +1,399 @@
+            { "referenceId", referenceId },
+            { "parameterName", inputConfig.Name ?? "" }
+        };
         input.SimulatorObjectReference = simulationObjectRef;
     }
 
+    private bool IsPlaceiTParameter(string referenceId)
+    {
+        // List of all known PlaceiT parameter reference IDs
+        var placeitParameters = new HashSet<string>
+        {
+            "porosity", "zonePressure", "zoneLength", "isothermOption", "isothermValues",
+            "adsorptionCapOption", "adsorptionCapValue", "enabledStages", "stagesVol",
+            "stagesConc", "injectionRate", "timestep", "productionTime", "productionRate", "medRange"
+        };
+        return placeitParameters.Contains(referenceId);
+    }
+
+    private void SetPlaceiTSimulationParameter(SimulatorRoutineRevisionInput inputConfig, SimulatorValueItem input, Dictionary<string, string> arguments)
+    {
+        // Use reference ID as the parameter key
+        var parameterName = inputConfig.ReferenceId ?? inputConfig.Name ?? "";
+        
+        object parameterValue = input.ValueType switch
+        {
+            SimulatorValueType.DOUBLE => (input.Value as SimulatorValue.Double)?.Value ?? 0.0,
+            SimulatorValueType.STRING => (input.Value as SimulatorValue.String)?.Value ?? "",
+            SimulatorValueType.DOUBLE_ARRAY => (input.Value as SimulatorValue.DoubleArray)?.Value?.ToArray() ?? Array.Empty<double>(),
+            SimulatorValueType.STRING_ARRAY => (input.Value as SimulatorValue.StringArray)?.Value?.ToArray() ?? Array.Empty<string>(),
+            _ => throw new NotImplementedException($"Value type '{input.ValueType}' not implemented for PlaceiT simulation")
+        };
+
+        _simulationParameters[parameterName] = parameterValue;
+        _logger.LogDebug($"Set PlaceiT parameter '{parameterName}' = {JsonSerializer.Serialize(parameterValue)}");
+    }
+
+    private void HandleComControlInput(SimulatorRoutineRevisionInput inputConfig, SimulatorValueItem input, Dictionary<string, string> arguments)
+    {
+        var action = (input.Value as SimulatorValue.String)?.Value ?? "";
+        
+        switch (action)
+        {
+            case "Initialize":
+                InitializePlaceiTComObject();
+                break;
+            case "Release":
+                ReleasePlaceiTComObject();
+                break;
+            case "Reset":
+                ResetSimulation();
+                break;
+            default:
+                throw new NotImplementedException($"COM control action '{action}' not implemented");
+        }
+    }
+
+    private void InitializePlaceiTComObject()
+    {
+        try
+        {
+            if (_placeitComObject == null)
+            {
+                // PlaceiT is implemented as VBA functions in the workbook
+                // The PlaceiTCOMEntryPoint function can be called directly on the workbook
+                _placeitComObject = _workbook;
```

Comment:
why create an extra variable `_placeitComObject` ? why not use `_workbook`directly?